presto-hudi/src/main/java/com/facebook/presto/hudi/HudiConnectorFactory.java

```diff
@@ -0,0 +1,104 @@
+import com.facebook.presto.spi.ConnectorHandleResolver;
+import com.facebook.presto.spi.NodeManager;
+import com.facebook.presto.spi.PageIndexerFactory;
+import com.facebook.presto.spi.classloader.ThreadContextClassLoader;
+import com.facebook.presto.spi.connector.Connector;
+import com.facebook.presto.spi.connector.ConnectorContext;
+import com.facebook.presto.spi.connector.ConnectorFactory;
+import org.weakref.jmx.guice.MBeanModule;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Throwables.throwIfUnchecked;
+import static java.util.Objects.requireNonNull;
+
+public class HudiConnectorFactory
+        implements ConnectorFactory
+{
+    private final String name;
+    private final ClassLoader classLoader;
+    private final Optional<ExtendedHiveMetastore> metastore;
+
+    public HudiConnectorFactory(
+            String name,
+            ClassLoader classLoader,
+            Optional<ExtendedHiveMetastore> metastore)
+    {
+        this.name = requireNonNull(name, "name is null");
+        this.classLoader = requireNonNull(classLoader, "classLoader is null");
+        this.metastore = requireNonNull(metastore, "metastore is null");
+    }
+
+    @Override
+    public String getName()
+    {
+        return name;
+    }
+
+    @Override
+    public ConnectorHandleResolver getHandleResolver()
+    {
+        return new HudiHandleResolver();
+    }
+
+    @Override
+    public Connector create(String catalogName, Map<String, String> config, ConnectorContext context)
+    {
+        requireNonNull(config, "config is null");
+
+        try (ThreadContextClassLoader ignore = new ThreadContextClassLoader(classLoader)) {
+            Bootstrap app = new Bootstrap(
+                    new EventModule(),
+                    new MBeanModule(),
+                    new JsonModule(),
+                    new HudiModule(classLoader, catalogName),
+                    new HiveS3Module(catalogName),
+                    new HiveAuthenticationModule(),
+                    new HiveMetastoreModule(catalogName, metastore),
+                    new CachingModule(),
+                    binder -> {
+                        binder.bind(NodeVersion.class).toInstance(new NodeVersion(context.getNodeManager().getCurrentNode().getVersion()));
+                        binder.bind(NodeManager.class).toInstance(context.getNodeManager());
+                        binder.bind(TypeManager.class).toInstance(context.getTypeManager());
+                        binder.bind(PageIndexerFactory.class).toInstance(context.getPageIndexerFactory());
+                    });
+
+            return app
+                    .doNotInitializeLogging()
+                    .setRequiredConfigurationProperties(config)
+                    .initialize()
+                    .getInstance(Connector.class);
+        }
+        catch (Exception e) {
+            throwIfUnchecked(e);
+            throw new RuntimeException(e);
```

Comment:
Do you want to include some additional context like catalogName or other information here.

Reply:
Fixed.